src/bitmessagekivy/mpybit.py

```diff
@@ -1,34 +1,171 @@
 
+    def __init__(self, **kwargs):
+        # pylint: disable=bad-super-call
+        super(OneLineListItem, self).__init__(**kwargs)
+        self.height = dp(16)
+
+
+class NavigationDrawerSubheader(OneLineListItem):
+    """
+    A subheader for separating content in :class:`MDNavigationDrawer`
+    Works well alongside :class:`NavigationDrawerDivider`
+    """
+
+    disabled = True
+    divider = None
+    theme_text_color = 'Secondary'
+
+
+class ContentNavigationDrawer(BoxLayout):
+    """ContentNavigationDrawer class for kivy Uir"""
+
+    def __init__(self, *args, **kwargs):
+        """Method used for contentNavigationDrawer"""
+        super(ContentNavigationDrawer, self).__init__(*args, **kwargs)
+        Clock.schedule_once(self.init_ui, 0)
+
+    def init_ui(self, dt=0):
+        """Clock Schdule for class contentNavigationDrawer"""
+        self.ids.scroll_y.bind(scroll_y=self.check_scroll_y)
+
+    def check_scroll_y(self, instance, somethingelse):
+        """show data on scroll down"""
+        if self.ids.btn.is_open:
+            self.ids.btn.is_open = False
+
+
+class CustomSpinner(Spinner):
+    """CustomSpinner class for kivy Ui"""
+
+    def __init__(self, *args, **kwargs):
+        """Method used for setting size of spinner"""
+        super(CustomSpinner, self).__init__(*args, **kwargs)
+        self.dropdown_cls.max_height = Window.size[1] / 3
+        self.values = list(addr for addr in config.addresses()
+                           if config.get(str(addr), 'enabled') == 'true')
+
+
+class NavigateApp(MDApp):
+    """Navigation Layout of class"""
     def __init__(self):
         super(NavigateApp, self).__init__()
         self.kivy_state_obj = KivyStateVariables()
 
-    def build(self):
-        """Method builds the widget"""
-        # pylint: disable=no-self-use
-        return Label(text="Hello World !")
+    title = "PyBitmessage"
+    image_path = os.path.join('./images', 'kivy')  # TODO: get path from kivy_state variable
 
-    def clickNavDrawer(self):
-        """method for clicking navigation drawer"""
-        pass
+    tr = Lang("en")  # for changing in franch replace en with fr
+
+    def build(self):  # pylint:disable=no-self-use
+        """Method builds the widget"""
+        for kv in data_screens:
+            Builder.load_file(
+                os.path.join(
+                    os.path.dirname(__file__),
+                    'kv',
+                    '{0}.kv'.format(all_data[kv]["kv_string"]),
+                )
+            )
+        return Builder.load_file(os.path.join(os.path.dirname(__file__), 'main.kv'))
 
-    def addingtoaddressbook(self):
-        """method for clicking address book popup"""
-        pass
+    def set_screen(self, screen_name):
+        """Set the screen name when navigate to other screens"""
+        self.root.ids.scr_mngr.current = screen_name
 
+    def run(self):
+        """Running the widgets"""
+        kivyuisignaler.release()
```

Comment:
if we redesign `kivyuisignaler` as a queue, then instead of
```kivyuisingaler.release()```
we can do something like
```kivyuisignaler.put("dummy")```

and wherever else you use ```kivyuisignaler.acquire()```, instead you'll use ```kivyuisignaler.get()```

I'm not 100% sure this isn't going to cause problems but this is approximately the approach to use.

Reply:
or you could use locks like `state.sql_ready`. This is probably cleaner than `put`/`get`, assuming the `kivyuisignaler` isn't used by anything else.

Reply:
Add event inside `kivy_state.py`
and used as `self.kivy_state_obj.sql_ready.set()`